themes/bootstrap3/templates/search/filters.phtml

```diff
@@ -0,0 +1,116 @@
+  $advancedSearch = $params->getSearchType() === 'advanced';
+?>
+<?php ob_start(); ?>
+  <?php foreach ($checkboxFilters as $filter): ?>
+    <?php if ($filter['selected']): ?>
+      <span class="filter-value">
+        <?php
+          $removeLink = isset($urlQuery) ? $urlQuery->removeFilter($filter['filter']) : '';
+          $desc = $this->translate($filter['desc']);
+          $ariaLabel = $this->translate('Remove filter') . ' ' . $this->escapeHtmlAttr($desc);
+        ?>
+        <?=$desc?>
+        <?php if($removeLink): ?>
+          <a aria-label="<?=$ariaLabel?>" href="<?=$removeLink?>"><i class="fa fa-window-close" aria-hidden="true"></i></a>
+        <?php endif; ?>
+      </span>
+    <?php endif ?>
+  <?php endforeach; ?>
+
+  <?php foreach ($filterList as $field => $data): ?>
+    <div class="title-value-pair">
+      <span class="filters-title"><?=$this->transEsc($field)?>:</span>
+      <?php foreach ($data as $index => $value): ?>
+        <?php
+          switch ($value['operator'])
+          {
+          case 'NOT':
+            $operatorChar = '-';
+            $join = $this->transEsc('NOT');
+            break;
+          case 'OR':
+            $operatorChar = '~';
+            $join = $this->transEsc('OR');
+            break;
+          default:
+            $operatorChar = '';
+            $join = '';
+          }
+          $operatorClass = $this->escapeHtmlAttr(strtolower($value['operator']));
+        ?>
+        <span class="filters-term filters-<?=$operatorClass?>">
+          <?=($index > 0 || 'NOT' === $value['operator']) ? $join : '' ?>
+        </span>
+        <span class="filter-value filters-<?=$operatorClass?>">
+          <?=$this->escapeHtml($value['displayText'])?>
+          <?php if (isset($this->urlQuery)): ?>
+            <?php $removeLink = $urlQuery->removeFacet($value['field'], $value['value'], $value['operator'])?>
+            <a aria-label="<?=$this->translate('Remove filter') ?>" href="<?=$removeLink?>"><i class="fa fa-window-close" aria-hidden="true"></i></a>
+          <?php endif; ?>
+        </span>
+        <?php if ($index == 0): ?></div><?php endif; ?>
+      <?php endforeach; ?>
+  <?php endforeach; ?>
+<?php $filters = ob_get_contents(); ?>
+<?php ob_end_clean(); ?>
+
+<?php if ($filterCount > 0): ?>
+  <?php // Normal view ?>
+  <div class="active-filters hidden-xs">
+    <?php if (empty($disableFilterSummary)): ?>
+      <?php if ($advancedSearch): ?>
+        <p class="adv_search_filters"><?=$this->transEsc('adv_search_filters')?>:</p>
+      <?php else: ?>
+        <div class="checkbox">
+          <label>
+            <input onChange="$('.active-filters .checkbox').toggleClass('checked'); var sort = $(this).closest('form').find('input[name=sort]'); sort.val($('.finna-filters .checkbox').hasClass('checked') ? sort.data('value') : '');" type="checkbox"<?=$defaultFilterState?> id="filter-checkbox" class="searchFormKeepFilters"/>
```

Comment:
I notice that there is some finna-specific logic here (and in another line below). Do you mind cleaning that up?

Reply:
Thanks for fixing this!